modules/coremark/core_portme.h

```diff
@@ -0,0 +1,206 @@
+ *
+ * Valid values :
+ * SEED_ARG - from command line.
+ * SEED_FUNC - from a system function.
+ * SEED_VOLATILE - from volatile variables.
+ */
+#ifndef SEED_METHOD
+  #define SEED_METHOD SEED_VOLATILE
+#endif
+
+/* Compiler information setup */
+#ifndef COMPILER_VERSION
+  #ifdef __GNUC__
+    #define COMPILER_VERSION "GCC"__VERSION__
+  #else
+    #define COMPILER_VERSION "Please put compiler version here (e.g. gcc 4.1)"
+  #endif
+#endif
+
+#ifndef COMPILER_FLAGS
+  #define COMPILER_FLAGS CONFIG_COMPILER_OPT
+#endif
+
+/* Definition of the method to get a block of memory */
+#ifndef MEM_METHOD
+  #ifdef CONFIG_COREMARK_MEMORY_METHOD_STACK
+    #define MEM_METHOD    MEM_STACK
+    #define MEM_LOCATION  "STACK"
+  #elif defined(CONFIG_COREMARK_MEMORY_METHOD_STATIC)
+    #define MEM_METHOD    MEM_STATIC
+    #define MEM_LOCATION  "STATIC"
+  #elif defined(CONFIG_COREMARK_MEMORY_METHOD_MALLOC)
+    #define MEM_METHOD    MEM_MALLOC
+    #define MEM_LOCATION  "MALLOC"
+  #endif
+#endif
+
+#ifndef MULTITHREAD
+  #define MULTITHREAD CONFIG_COREMARK_THREADS_NUMBER
+#endif
+
+#if (MULTITHREAD > 1)
+  #define PARALLEL_METHOD "PThreads"
+#endif
+
+/* Depending on the benchmark run type different data size is required.
+ * Default values for TOTAL_DATA_SIZE is predefined in coremark.h
+ * Redefine it here and get value according to coremark configuraion.
+ */
+#undef TOTAL_DATA_SIZE
+#define TOTAL_DATA_SIZE  CONFIG_COREMARK_DATA_SIZE
+
+/* CoreMark's entry point, called 'main' by default.
+ * As CoreMark sources are read-only this conflict must be solved in other way.
+ * To use it, the CoreMark main function needs to be renamed.
+ */
+#define main       coremark_run
+
+/* The crc16 function is present both in Zephyr and CoreMark.
+ * To avoid multiple definition error without changing benchmark's code,
+ * coremark crc16 instance is renamed.
+ */
+#define crc16      coremark_crc16
+
+/* This function will be used to output benchmark results */
+#define ee_printf  printk
+
+/* Functions used by CoreMark for MEM_METHOD MALLOC case */
+#define portable_malloc  k_malloc
+#define portable_free    k_free
+
+#define THREAD_STACK_SIZE   (512)
+
+#define _COREMARK_THREAD_STACK_DEFINE(n, _)					 \
+	static K_THREAD_STACK_DEFINE(CONCAT(CONCAT(coremark_thread_, n), _stack),\
+				     THREAD_STACK_SIZE)
+
+#define _COREMARK_THREAD_STACK_ARRAY_ITEM(n, _) CONCAT(CONCAT(coremark_thread_, n), _stack)
+
+/**
+ *  @brief Statically define thread stack structure array.
+ *
+ *  Helper macro to statically define thread stack structure array.
+ *
+ *  @param _name	 Name of stack structure array.
+ *  @param _instance_num Number of elements in instance array.
+ */
+#define COREMARK_THREAD_STACK_INSTANCE_DEFINE(_name, _instance_num)		 \
+	LISTIFY(_instance_num, _COREMARK_THREAD_STACK_DEFINE, (;));		 \
+	static k_thread_stack_t *_name[] = {					 \
+		LISTIFY(_instance_num, _COREMARK_THREAD_STACK_ARRAY_ITEM, (,))	 \
+	}
```

Comment:
These macros are only intended to be used in the cort_portme.c file. Can we define them there (at the top of the file)?